client/src/components/DashboardHeader/DashboardHeader.tsx

```diff
@@ -12,6 +21,22 @@ export default function DashboardHeader(): JSX.Element {
 
   const loggedInUser: User = { email: 'rose@gmail.com', username: 'Rose' }; //temp
 
+  const { logout } = useAuth();
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const popoverOpen = Boolean(anchorEl);
+  const popoverId = popoverOpen ? 'user-popover' : undefined;
```

Comment:
Is there a reason you'd want the id to be undefined at some points rather than it always being 'user-popover'?

Reply:
No worries! I just noticed that in the MUI docs as well, not sure why they do that. Up to you if you want to change it, but go ahead and do what they do if something breaks!